quantum/rgb_matrix_types.h

```diff
@@ -79,12 +79,11 @@ typedef struct PACKED {
 typedef union {
   uint32_t raw;
   struct PACKED {
-    uint8_t  enable  :2;
-    uint8_t  mode    :6;
-    uint8_t  hue     :8;
-    uint8_t  sat     :8;
-    uint8_t  val     :8;
-    uint8_t  speed   :8;//EECONFIG needs to be increased to support this
+    uint8_t  mode;
```

Comment:
The only issue with removing `enable`, is that it looks like there is no longer a way to save having the lights disabled?  

If so, that's a problem.

Reply:
Ya, so that is something I was curious as to people's thoughts. Right now, Enable is saved, however Speed is not, also nothing is saved on Arm. Also with multiple effect layers, only the first layer is saved. So there is a ton that won't be saved without adding a lot to the eeprom saving system. 

So, is it worth doing the work and getting everything saving? (enable, speed, layers, etc)
Should that be in this pr or a separate pr?

Reply:
At the very least, enable should be saved.  

And you're half right, EEPROM does work on ChibiOS based ARM boards.  But not ATSAM, as far as I'm aware.

As for how to handle the multiple zones, I'm not sure.